feat: add dev.pulsar_edit.Pulsar.metainfo.xml

cat-master21 opened the pull request:
This adds metainfo for Linux distributions. This is not packaged in anything yet, but it may be utilized be Flatpak, Deb, Rpm, etc.

Daeraxa commented:
I fear this is going to get quickly forgotten and really wouldn't we want an action or something to update this file automatically because this just adds more to the release overhead.

```
<releases>
  <release version="1.101.0" date="2023-01-15"/>
</releases>
```

cat-master21 commented:
Alright then I have removed the releases section then

confused-Techie commented:
This seems like it could be helpful once we start packaging on those platforms.

My only notes would be how do we feel about the author being called "Pulsar Team", I think in the past we've gone more for Pulsar-Edit or just Pulsar.

The description is fine as that's what's on our org readme, but wondering if anyone has something that might be more succinct? 

Finally for the metadata license are we able to assign that as MIT? Just so it matches the project license, if not fair enough but was just curious.

Meadowsys commented:
hmm.. not sure how I feel about the description either?

cat-master21 commented:
> This seems like it could be helpful once we start packaging on those platforms.
> 
> My only notes would be how do we feel about the author being called "Pulsar Team", I think in the past we've gone more for Pulsar-Edit or just Pulsar.
> 
> The description is fine as that's what's on our org readme, but wondering if anyone has something that might be more succinct?
> 
> Finally for the metadata license are we able to assign that as MIT? Just so it matches the project license, if not fair enough but was just curious.

So the license for metadata can be MIT and I've changed the description to `Pulsar-edit` (please update https://pulsar-edit.dev/docs/launch-manual/sections/faq/#having-trouble to Pulsar-edit by the way). As for the description, I removed the second sentence if that helps.

confused-Techie commented:
> > This seems like it could be helpful once we start packaging on those platforms.
> > My only notes would be how do we feel about the author being called "Pulsar Team", I think in the past we've gone more for Pulsar-Edit or just Pulsar.
> > The description is fine as that's what's on our org readme, but wondering if anyone has something that might be more succinct?
> > Finally for the metadata license are we able to assign that as MIT? Just so it matches the project license, if not fair enough but was just curious.
> 
> So the license for metadata can be MIT and I've changed the description to `Pulsar-edit` (please update https://pulsar-edit.dev/docs/launch-manual/sections/faq/#having-trouble to Pulsar-edit by the way). As for the description, I removed the second sentence if that helps.

I really appreciate you being so receptive to everyone's comments! And of course appreciate the contribution!

You are right that we have used `Pulsar Team` a few times so as you can tell that terminology is still up in the error. But yeah what you've now put feels better to me.

As for everything else this looks good to me, description feels still a tad verbose, but I don't think that's an issue of what you've put, more of what we have put out there, so we can talk about that elsewhere and leave your PR alone, unless someone can think of something actionable to say about it. Otherwise this looks fine by me!

Meadowsys commented:
should the capitalisation of "Pulsar-edit" be instead "Pulsar-Edit"? or is it good as is?

cat-master21 commented:
> should the capitalisation of "Pulsar-edit" be instead "Pulsar-Edit"? or is it good as is?

Looking at the GitHub organization readme, it looks like it is spelled "Pulsar Edit" so I have updated the author name to that.

confused-Techie reviewed (approved):
At this point all of my concerns have been addressed, and this looks fantastic!

DeeDeeG commented:
Just wanted to post some related docs in this PR so people can easily find them:

- https://docs.flatpak.org/en/latest/freedesktop-quick-reference.html#appdata-files
- https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

I assume this file is most immediately relevant to the FlatHub packaging effort. Since that already has the file exactly as it was proposed at the beginning of this PR: https://github.com/flathub/dev.pulsar_edit.Pulsar/blob/master/dev.pulsar_edit.Pulsar.metainfo.xml. So this would basically upstream that one metadata file into our project, rather than have it solely maintained in the FlatHub packaging repo. Seems like a decent first step to bringing the FlatHub packaging in-house and making it official, should we eventually want to do that. (I have heard folks say in the Discord that we do, eventually, perhaps soon.)

(I haven't gotten around to properly reviewing this yet... And may not get to it, so other reviewers: don't wait for me if you feel you're ready to merge, I'm not trying to slow anything down, just had time to add some context and not do much else at the moment, sorry!)

DeeDeeG commented on resources/linux/dev.pulsar_edit.Pulsar.metainfo.xml:
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>dev.pulsar_edit.Pulsar</id>
```
Are hyphens (`-`) not allowed in the `<id>` field for flatpak/flathub packages?

If hyphens are allowed, if this isn't some technically necessary workaround, then this should ideally rather use a hyphen (`-`) than an underscore (`_`).

DeeDeeG replied:
**NOTE**: If changing this means the old `<id>` will still exist over at FlatHub, and not match, we may have to weigh continuity and the risk of someone mis-using the old id, with having the id match the convention we have set up around the project already.

confused-Techie replied:
Without docs I had assumed this was limitation or convention over on flatpak. But if they are allowed ideally I'd say we use the general convention. Although if the old now in use ID can't be deleted we could always keep using this id until we officially package and distribute Pulsar (if that happens) and at that time we then use our general conventions when creating the ID.

That would mean there are two semi legit ones, but I'm not sure how much of a concern that really would be.

cat-master21 replied:
Yeah my bad that was before I fixed that for flathub.

Meadowsys replied:
<https://github.com/flathub/flathub/pull/3850#discussion_r1086180488>

It was asked to be changed to an underscore during submission to flathub

DeeDeeG replied:
I... I think the D-Bus conventions (and the FlatPak conventions based on them) are really weird an counter-intuitive, but this actually does follow those conventions perfectly, on closer reading.

Thanks for linking that pr @Meadowsys, following the discussion there was very helpful for me!

I'm going to mark this as resolved... It's correct as-is! And the FlatHub folks said so themselves!

DeeDeeG reviewed (approved):
Approving.

I've been made aware that this was already approved as being technically done right by the FlatHub reviewers over in https://github.com/flathub/flathub/pull/3850.

And our own suggestions have been addressed.

I can't think of anything wrong with it. The description is kinda interesting, dunno if it's exactly what I'd put. but I think we can change that later on, and I don't immediately have any suggestions to improve it.

That all being said, LGTM. Happy to upstream this to our repo.

confused-Techie commented:
With two (sorta 3 if you include the FlatHub reviewers` I'll go ahead and merge this one.

As always appreciate you contributing!
